### PR TITLE
Update AndroidComponentAddress to wrap a binding Intent.

### PR DESCRIPTION
**Pull request by jdcormie**
By considering the binding Intent's action, data, type, identity and categories we align gRPC/Binder's addressing with Android's natural equivalence relation for "cached" IBinders.

### CODE REVIEW

File: binder/src/main/java/io/grpc/binder/BinderServerBuilder.java [hunk 1]
```diff
@@ -53,17 +53,35 @@
...
+   * @param receiver an "out param" for the new {@link Server}'s listening {@link IBinder}
+   * @return a new builder
+   */
+  public static BinderServerBuilder forAddress(AndroidComponentAddress listenAddress,
+      IBinderReceiver receiver) {
+    return new BinderServerBuilder(listenAddress, receiver);
+  }
+
+  /**
+   * Creates a server builder that will listen for bindings to the specified Service with the
```
Comment (markb74) [inline binder/src/main/java/io/grpc/binder/BinderServerBuilder.java:72]: Let's just remove this method since it doesn't serve much purpose anymore.

**Review by markb74 — APPROVED** [review]

File: binder/src/main/java/io/grpc/binder/BinderServerBuilder.java (hunk 1 above)
Comment (jdcormie) [reply binder/src/main/java/io/grpc/binder/BinderServerBuilder.java]: Done.

**Comment by markb74** [comment]
There are a couple of tests which use BinderServerBuilder.forService you'll need to update.

**Comment by jdcormie** [comment]
> There are a couple of tests which use BinderServerBuilder.forService you'll need to update.

Oops I didn't realize there were tests not run by "gradlew test". Thanks and done.

**Pull request merged**